Add a PerfKitExplorer dashboard for every benchmark.

yuyantingzero opened the pull request.

cmccoy commented:
Would you mind:

* Pretty-printing these ([`jq`](http://stedolan.github.io/jq/) does a fine job, as does Python's `json.dumps(..., indent=2)`).
* Naming each JSON file after the benchmark it tracks.

yuyantingzero commented:
PTAL

voellm commented on tools/dashboard/README.md:
```diff
@@ -0,0 +1,17 @@
+`https://github.com/GoogleCloudPlatform/PerfKitExplorer` for How to setup
+PerfKitExplorer.
+
+## Setup dashboards
```
I dont see how this describes what the user needs to do to deploy the dashboards to the explorer.   The sed commands wont really do anything as-is.

yuyantingzero replied:
Added more details in root README and dashboards/README

voellm commented:
In addition to the readme.md in the dashboards directory I think you need to have help in the top level readme.md as well.

Also I think the dashboards should be rooted at the top level vs under tools.

cmccoy commented:
I like them under `/tools` - keeping the top-level directory uncluttered is nice.

yuyantingzero commented:
Moved to root directory and renamed the folder to dashboards. Since the next step will be adding integration tests for benchmarks. This will be more than a tool.

cmccoy commented on README.md:
```diff
@@ -377,6 +377,24 @@ file we have added a lot of comments into the code to make it easy to;
+(https://github.com/GoogleCloudPlatform/PerfKitExplorer/blob/master/data/samples_mart/results_table_schema.json)
+file is needed):
+
+        bq load --source_format=NEWLINE_DELIMITED_JSON BIGQUERY_PROJECT_ID:BIGQUERY_DATASET.BIGQUERY_TABLE RESULTS_FILE (BIGQUERY_SCHEMA)
```
Any reason for preferring this over the `pkb` flags `--bigquery_table` / `--bq_project`?

yuyantingzero replied:
The flags only works if the table already exists. Added the flag option and mentioned that it only works if the table exists.

cmccoy commented on dashboards/README.md:
```diff
@@ -0,0 +1,18 @@
+
+See:
+
+`https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/README.md` for setting up PerfKitExplorer and load data into Google BigQuery.
```
Nit: use a markdown link here.

yuyantingzero replied:
Done.

yuyantingzero commented:
PTAL

voellm commented:
By the way I am OK with this under tools but my vote is at the top layer.

The reason I don't prefer tools is that these are not a tool... the are a
feature set.
On Jun 23, 2015 10:48 PM, "yuyanting" <notifications@github.com> wrote:

> PTAL
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/332#issuecomment-114637794>
> .
>

cmccoy commented on README.md:
```diff
@@ -377,6 +377,29 @@ file we have added a lot of comments into the code to make it easy to;
 Even with lots of comments we make to support more detailed documention.  You will find the documatation we have on the [Wiki pages] (https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/wiki).  Missing documentation you want?  Start a page and/or open an [issue] (https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/issues) to get it added.
 
+
+HOW TO VISUALIZE PerfKitBenchmarker RESULTS
```
PerfKitExplorer isn't the *only* way to visualize the results. Maybe make this "Visualizing the results in PerfKitExplorer"?

cmccoy commented on README.md:
```diff
@@ -377,6 +377,29 @@ file we have added a lot of comments into the code to make it easy to;
+
+HOW TO VISUALIZE PerfKitBenchmarker RESULTS
+==================
+The dashboards directory contains a list of pre-built dashboards for every benchmark in PerfKitBenchmarker and a summary dashboard for all benchmarks. The dashboards can be used by PerfKitExplorer.
```
I would recommend keeping this section very short - maybe just this sentence and your "see  `dashboards/README.md`" link.

cmccoy replied:
(and moving anything only covered here to `dashboards/README.md`).

cmccoy commented on dashboards/README.md:
```diff
@@ -0,0 +1,18 @@
+        sed -i  's/{{ dataset_name }}/BIGQUERY_DATASET/g' *.json
+        sed -i  's/{{ table_name }}/BIGQUERY_TABLE/g' *.json
+
+* Go to PerfKitExplorer site deployed by AppEngine: http://version-dot-EXPLORER_PROJECT_ID.appspot.com
```
Grammar: "Go to the" and deployed by -> deployed on.

cmccoy commented on dashboards/README.md:
```diff
@@ -0,0 +1,18 @@
+        sed -i  's/{{ table_name }}/BIGQUERY_TABLE/g' *.json
+
+* Go to PerfKitExplorer site deployed by AppEngine: http://version-dot-EXPLORER_PROJECT_ID.appspot.com
+* Click "Upload" button in the "PerfKit Dashboard Administration" to upload the dashboards.
```
Could you clarify this? (in the "PerfKit Dashboard Administration"...menu? Section?)

cmccoy commented on dashboards/README.md:
```diff
@@ -0,0 +1,18 @@
+# README
+
+This folder contains a list of pre-built dashboards for every benchmark in PerfKitBenchmarker and a summary dashboard for all benchmarks. The dashboards can be used by PerfKitExplorer.
```
remove "a list" - you have teh dashboards themselves.
Link to perfkitexplorer here.

    [PerfKitExplorer](http://github.com/GoogleCloudPlatform/PerfKitExplorer)

cmccoy commented on dashboards/README.md:
```diff
@@ -0,0 +1,18 @@
+for setting up PerfKitExplorer and load data into Google BigQuery.
+
+## Setup dashboards
+* Once the PerfKitExplorer is deployed, config the the dashboards with your BIGQUERY_PROJECT_ID, BIGQUERY_DATASET and BIGQUERY_TABLE used by PerfKitExplorer by running the following command in this directory:
```
the PerfKitExplorer -> PerfKitExplorer
config -> configure

voellm commented:
This CL is going to move to the PerfKitExplorer project.
